### ISSUE TITLE
[Bug]  Messaging Subscribe stays pending on Android

### ISSUE THREAD

**Issue by Neitah** [firebase/firebase-cpp-sdk#1361]
### [REQUIRED] Please fill in the following fields:

  * Pre-built SDK from the [website](https://firebase.google.com/download/cpp) or open-source from this repo: Pre-built SDK
  * Firebase C++ SDK version: 11.2.0
  * Problematic Firebase Component: Messaging
  * Other Firebase Components in use: Core
  * Platform you are using the C++ SDK on: Windows 
  * Platform you are targeting: Android

### [REQUIRED] Please describe the issue here:

Created a Qt 6.3.1 project with Firebase Messaging.

Received the Firebase Authentication Token

Tried to subscribe to a topic, and it stays pending.

The same code works on iOS.

#### Steps to reproduce:

Have you been able to reproduce this issue with just the [Firebase C++ quickstarts](https://github.com/firebase/quickstart-cpp) ?
What's the issue repro rate? (eg 100%, 1/5 etc)
100%

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.
The status of the Future gotten from Subscribe stays as kFutureStatusPending

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:

Token received like described here : 
https://medium.com/@shigmas/push-notifications-in-qt-for-ios-and-android-50c2f9a38444

Then, I just did this : 

```::firebase::messaging::Subscribe("TopicToSubscribeObstructed");```

**Comment by a-maurice** [comment]
I'm not clear what the problems you are seeing might be, but there is a handful of additional steps that are needed for Android, such as updating the AndroidManifest to use the ListenerService and MessageForwardingService.  For more instructions you can see: https://firebase.google.com/docs/cloud-messaging/cpp/client

For more examples, you can also see https://github.com/firebase/quickstart-cpp/tree/main/messaging/testapp which includes more set up instructions as well.

**Comment by Neitah** [comment]
Well, the thing is, I receive notifications just fine when using the device token to send, so I don't think that's the problem ? Unless there's additional things to put in the manifest for topic messaging (which I didn't see), then that might be where it comes from ^^'

**Comment by a-maurice** [comment]
Hmm, I can't find anything that would be necessary for subscribing to topics that you wouldn't need for other steps as well.  Do you see anything useful in the logs?  You can also adjust the log level via:
```
#include "firebase/log.h"

firebase::SetLogLevel(firebase::kLogLevelVerbose);
```
which might indicate what the problem is.

**Comment by ChristophePretty** [comment]
Hi,

I have the same issue on my side, and it is not solved by release 11.4.0:
- I am trying to subscribe to a topic, on IOS it is working great and my completion callback is called, on Android, the completion callback is never called. 
(FYI, until the bug is fixed, I am using the Java SDK just for the subscribe and it is working)

Thanks
